### PR TITLE
fix: file methods and GlobalMiddleware method

### PR DESCRIPTION
**Pull request by devhaozi**
Closes goravel/goravel#299
Closes goravel/goravel#300

## 📑 Description

Fix File methods not work.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: group_test.go [hunk 1]
```diff
@@ -384,24 +386,45 @@ func TestGroup(t *testing.T) {
...
+				tempDir, err := os.MkdirTemp("", "test")
+				assert.NoError(t, err)
+				defer os.RemoveAll(tempDir)
+
+				err = os.WriteFile(filepath.Join(tempDir, "test.json"), []byte("{\"id\":1}"), 0755)
+				assert.NoError(t, err)
+
 				fiber.StaticFS("static-fs", http.Dir("./"))
 			},
 			method:     "GET",
```
Comment (hwbrzzl) [inline group_test.go:430]: The URL of next line is `/static-fs/README.md`, do we need to change it？

Comment (hwbrzzl) [reply group_test.go:430]: Oh, we may don't need to change this test case, it's `http.Dir("./")`.

Comment (devhaozi) [reply group_test.go]: Fixed

**Comment by hwbrzzl** [comment]
CI failed

**Comment by devhaozi** [comment]
> CI failed

Fixed

**Review by hwbrzzl — APPROVED** [review]
LGTM

**Comment by frank505** [comment]
i am having an issue with uploading files together with strings using multipart formdata here is my definitions 
```
projectId := ctx.Request().Input("projectId")
	organizationId := ctx.Request().InputInt64("organizationId")
	removeBranding := ctx.Request().InputBool("remove_branding")
	textColor := ctx.Request().Input("text_color")
	playBackMode := ctx.Request().Input("playback_mode")
	thumbnail, _ := ctx.Request().File("thumbnail")
	mimeType, _ := thumbnail.MimeType()
	validator, err := facades.Validation().Make(
		map[string]any{
			"organizationId": organizationId,
			"projectId":      projectId,
			"removeBranding": removeBranding,
			"textColor":      textColor,
			"thumbnail":      mimeType,
			"playBackMode":   playBackMode,
		},
		map[string]string{
			"organizationId": "required|number",
			"projectId":      "required|string",
			"removeBranding": "boolean",
			"textColor":      "string",
			"thumbnail":      "string",
			"playBackMode":   "string",
		},
		validation.Messages(
			map[string]string{
				"organizationId.required": "Organization ID is required",
				"projectId.required":      "Project ID is required",
				"organizationId.number":   "Organization ID must be a number",
				"projectId.string":        "Project ID must be a string",
				"removeBranding.boolean":  "Remove Branding must be true or false",
				"textColor.string":        "Text Color must be a string",
				"thumbnail.string":        "required|string|in:image/jpeg,image/png,image/gif,image/bmp,image/svg+xml,image/webp",
				"playBackMode.string":     "PlayBack Mode must be a valid JSON string",
			},
		),
	)

	if err != nil {
		return ctx.Response().Status(400).Json(
			http.Json{
				"errors":  err.Error(),
				"success": false,
			})
	}

	if validator.Fails() {
		return ctx.Response().Status(400).Json(
			http.Json{
				"success": false,
				"errors":  validator.Errors().All(),
			})
	}
``` it works fine if i am uploading only a file but together with other values does not work as the other values are returning the default of the Input functions rather than the values passed

**Pull request merged**